File: backend/src/main/java/com/ll/hotel/global/aspect/ServiceLoggingAspect.java
package com.ll.hotel.global.aspect;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.aspectj.lang.JoinPoint;
import org.aspectj.lang.annotation.Aspect;
import org.aspectj.lang.annotation.Before;
import org.springframework.stereotype.Component;

@Aspect
@Component
@Slf4j
@RequiredArgsConstructor
public class ServiceLoggingAspect {
    @Before("@within(org.springframework.stereotype.Service)")
    public Object logService(JoinPoint joinPoint) {
        String className = joinPoint.getSignature().getDeclaringType().getSimpleName();
        String methodName = joinPoint.getSignature().getName();
        log.info("Request Service = [{}.{}]", className, methodName);

        return joinPoint;
    }
}

File: backend/src/main/java/com/ll/hotel/global/aspect/ServiceLoggingAspect.kt
package com.ll.hotel.global.aspect

import com.ll.hotel.standard.util.LogUtil
import com.ll.hotel.standard.util.logger
import org.aspectj.lang.JoinPoint
import org.aspectj.lang.annotation.Aspect
import org.aspectj.lang.annotation.Before
import org.springframework.stereotype.Component

@Aspect
@Component
class ServiceLoggingAspect {
    private val log = logger()

    @Before("@within(org.springframework.stereotype.Service)")
    fun logService(joinPoint: JoinPoint): Any {
        val className = joinPoint.signature.declaringType.simpleName
        val methodName = joinPoint.signature.name

        LogUtil.logServiceRequest(log, className, methodName)

        return joinPoint
    }
}
